### PR TITLE
feat: new type scale and font

### PR DESCRIPTION
**Pull request by MuckT**
### Description

Adds a new typography for fonts going forward, marks existing fontStyles as deprecated and adds a new font `Inter-Bold`.
### Related issues

- Fixes RET-740

### Backwards compatibility

Yes

### CODE REVIEW

File: src/styles/fonts.tsx [hunk 1]
```diff
@@ -40,8 +41,110 @@ const standards = {
...
+
+/**
+ * Figma TypeScale Styles
+ */
+export const typeScale = StyleSheet.create({
+  displayLarge: {
+    fontFamily: Inter.Bold,
+    fontSize: 80,
+    lineHeight: 80,
+    letterSpacing: -3,
```
Comment (jeanregisser) [inline src/styles/fonts.tsx:53]: Looking at the figma file and selecting this one, it says `letter-spacing: -2.4px;` as CSS prop. Which is 3% from 80 (the line height).

Screenshot 2023-09-20 at 10 33 38

```suggestion
    letterSpacing: -2.4,
```

Similar adjustments are needed for the other styles below.

**Review by jeanregisser — COMMENTED** [review]
Great! 👍

**Comment by bakoushin** [comment]
Finally! 🚀

File: src/styles/fonts.tsx (hunk 1 above)
Comment (MuckT) [reply src/styles/fonts.tsx:53]: Good catch I'll make this adjustment!

Comment (MuckT) [reply src/styles/fonts.tsx]: Fixed in https://github.com/valora-inc/wallet/pull/4211/commits/9dc5d3e480cbd810c3c23eaea219f22401219f7c!

File: src/styles/fonts.tsx [hunk 2]
```diff
@@ -40,8 +41,110 @@ const standards = {
...
+  labelSmall: {
+    fontFamily: Inter.Medium,
+    fontSize: 14,
+    lineHeight: 20,
+  },
+  labelXSmall: {
+    fontFamily: Inter.Medium,
+    fontSize: 12,
+    lineHeight: 16,
+    letterSpacing: 0.16,
```
Comment (satish-ravi) [inline src/styles/fonts.tsx:102]: should this be 0.12? Seems like everything else is a % of `fontSize`, except this which seems to be a % of `lineHeight`

Comment (MuckT) [reply src/styles/fonts.tsx]: I think it should be too on second glance based on what I see in Figma: % appears related to the fontSize instead lineHeight. Fixed in https://github.com/valora-inc/wallet/pull/4211/commits/eae1ba9ac015d041d7168da7b4301c6ca297b0cf.

**Review by satish-ravi — APPROVED** [review]

**Pull request merged**